commons/src/test/java/org/openmrs/eip/BaseCamelTest.java

```diff
@@ -69,76 +69,54 @@ public abstract class BaseCamelTest {
 	protected ConfigurableEnvironment env;
 	
 	protected void advise(String routeId, AdviceWithRouteBuilder builder) throws Exception {
-		camelContext.adviceWith(camelContext.getRouteDefinition(routeId), builder);
+		adviceWith(routeId, camelContext, builder);
 	}
 	
-	@Before
+	@BeforeEach
 	public void beforeBaseCamelTest() throws Exception {
-		loadXmlRoutesInDirectory("camel-common", "test-error-handler.xml");
 		if (loggerContext == null) {
 			loggerContext = (LoggerContext) LoggerFactory.getILoggerFactory();
 		}
 		
-		((ListAppender) loggerContext.getLogger(ROOT_LOGGER_NAME).getAppender("test")).list.clear();
+		((ListAppender<?>) loggerContext.getLogger(ROOT_LOGGER_NAME).getAppender("test")).list.clear();
 	}
 	
+	/**
+	 * Asserts that the exchange has failed
+	 *
+	 * @param level   the log level
+	 * @param message the log message
+	 */
 	protected void assertMessageLogged(Level level, String message) {
+		log.info("Checking for log event -> [" + level + "] " + message);
 		ListAppender<LoggingEvent> app = (ListAppender) loggerContext.getLogger(ROOT_LOGGER_NAME).getAppender("test");
 		List<LoggingEvent> list = app.list;
-		for (LoggingEvent e : list) {
-			if (e.getLevel().equals(level) && e.getMessage().equalsIgnoreCase(message)) {
+		for (LoggingEvent event : list) {
+			if (event.getLevel().equals(level) && event.getMessage().equalsIgnoreCase(message)) {
 				log.info("Log event satisfied -> [" + level + "] " + message);
 				return;
 			}
 		}
-		
-		Assert.fail("Log event not satisfied -> [" + level + "] " + message);
 	}
 	
 	/**
-	 * Loads and registers the routes defined in the files on the classpath with the specified names.
+	 * Returns the body of the exchange
 	 *
-	 * @param filenames the names of the files to load
-	 * @throws Exception
-	 */
-	private void loadXmlRoutes(String... filenames) throws Exception {
-		for (String file : filenames) {
-			InputStream in = getClass().getClassLoader().getResourceAsStream(file);
-			RoutesDefinition rd = (RoutesDefinition) camelContext.getXMLRoutesDefinitionLoader()
-			        .loadRoutesDefinition(camelContext, in);
-			camelContext.addRouteDefinitions(rd.getRoutes());
-		}
-	}
```

Comment:
This `loadXmlRoutes(String... filenames)` method is used by test classes downstream, but we can replicate the same functionality. However, I advise against this, as the `XMLRoutesDefinitionLoader` class is deprecated in favor of generic loaders in this [PR](https://github.com/apache/camel-spring-boot/pull/276). By default, loads from the classpath.

To load XML routes in your code, use the following property:

```xml
camel.springboot.routes-include-pattern=file:src/test/resources/routes/foo.xml,file:src/test/resources/routes/bar.xml
```